docs/declaring_rbac_rules_for_controllers.md

```diff
@@ -1,30 +0,0 @@
-# Declaring RBAC rules for controllers
```

Comment:
The content in this file seems to still be valid.
Do we have a replacement somewhere else?

Reply:
Good catch. Part of the documentation is valid in this file. Filed an issue for 
https://github.com/kubernetes-sigs/kubebuilder/issues/426 for adding new section in kubebuilder book.

Reply:
I'm fine with doing it in a future PR :)